commands/src/main/java/org/wildfly/extras/creaper/commands/ra/AddConnectionFactoryToRA.java

```diff
@@ -0,0 +1,147 @@
+        if (password != null) {
+            ops.writeAttribute(address, "recovery-password", password);
+        }
+        ops.writeAttribute(address, "tracking", tracking);
+        if (properties.size() > 0) {
+
+            Iterator it = properties.entrySet().iterator();
+            while (it.hasNext()) {
+                Map.Entry pair = (Map.Entry) it.next();
+                Address configPropsAddress = Address.subsystem("resource-adapters")
+                        .and("resource-adapter", resourceAdapterId)
+                        .and("connection-definitions", poolName).and("config-properties", (String) pair.getKey());
+                ops.add(configPropsAddress, Values.of("value", (String) pair.getValue()));
+            }
+
+        }
+
+
+    }
+
+    public static final class Builder {
+        private String poolName;
+        private String localJndiName;
+        private String resourceAdapterId;
+        private String userName = null;
+        private String password = null;
+        private boolean security = false;
+        private boolean tracking = true;
+        private Map<String, String> properties = new HashMap<String, String>();
+
+        /**
+         * @param poolName          unique id for CF (not used for jndi lookup)
+         * @param localJndiName     jndi name for CF used for jndi lookup from apps
+         * @param resourceAdapterId id of the RA where this CF will be added
+         */
+        public Builder(String poolName, String localJndiName, String resourceAdapterId) {
+            if (poolName == null) {
+                throw new IllegalArgumentException("poolName must be specified");
+            }
+            if (localJndiName == null) {
+                throw new IllegalArgumentException("localJndiName must be specified");
+            }
+            if (resourceAdapterId == null) {
+                throw new IllegalArgumentException("resourceAdapterId must be specified");
+            }
+            this.poolName = poolName;
+            this.localJndiName = localJndiName;
+            this.resourceAdapterId = resourceAdapterId;
+        }
+
+        /**
+         * XA recovery credentials
+         *
+         * @param userName user name
+         * @param password password
+         */
+        public Builder addXARecovery(String userName, String password) {
+            this.userName = userName;
+            this.password = password;
+            return this;
+        }
+
+        /**
+         * @param applicationSecurity sets security element if true
+         */
+        public Builder setAppliactionSecurity(boolean applicationSecurity) {
+            this.security = applicationSecurity;
+            return this;
+        }
+
+        /**
+         * Other properties depending on particular  RA requirements
+         *
+         * @param name  name of property
+         * @param value value
+         */
+        public Builder addCustomConfigProperty(String name, String value) {
```

Comment:
Does the subsystem really use the name "custrom config property"? If not, simple `addProperty` would be enough.